### ISSUE TITLE
forward slash on table name causes ORA-01424 and blocks the sync step

### ISSUE THREAD

**Issue by geraldodev** [metabase/metabase#13225]
**Describe the bug**
Databases with forward slash on table name causes ORA-01424 and breaks the sync database step.

**Logs**
```
{
  "browser-info": {
    "language": "pt-BR",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_265-8u265-b01-0ubuntu2~18.04-b01",
    "java.vendor": "Private Build",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_265",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.265-b01",
    "os.name": "Linux",
    "os.version": "4.15.0-111-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "oracle"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.3 (Ubuntu 12.3-1.pgdg18.04+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.36.4",
      "date": "2020-08-17",
      "branch": "release-0.36.x",
      "hash": "196c1f6"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```
```
java.sql.SQLDataException: ORA-01424: missing or illegal character following the escape character

        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:494)
        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:446)
        at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1054)
        at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:623)
        at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:252)
        at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:612)
        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:226)
        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:59)
        at oracle.jdbc.driver.T4CPreparedStatement.executeForDescribe(T4CPreparedStatement.java:747)
        at oracle.jdbc.driver.OracleStatement.executeMaybeDescribe(OracleStatement.java:904)
        at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1082)
        at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3780)
        at oracle.jdbc.driver.T4CPreparedStatement.executeInternal(T4CPreparedStatement.java:1343)
        at oracle.jdbc.driver.OraclePreparedStatement.executeQuery(OraclePreparedStatement.java:3822)
        at oracle.jdbc.driver.OraclePreparedStatementWrapper.executeQuery(OraclePreparedStatementWrapper.java:1165)
        at oracle.jdbc.OracleDatabaseMetaData.getTables(OracleDatabaseMetaData.java:2721)
        at com.mchange.v2.c3p0.impl.NewProxyDatabaseMetaData.getTables(NewProxyDatabaseMetaData.java:3010)
        at metabase.driver.sql_jdbc.sync$get_tables.invokeStatic(sync.clj:103)
        at metabase.driver.sql_jdbc.sync$get_tables.invoke(sync.clj:99)
        at metabase.driver.sql_jdbc.sync$fast_active_tables$iter__72128__72134$fn__72135.invoke(sync.clj:119)
        at clojure.lang.LazySeq.sval(LazySeq.java:42)
        at clojure.lang.LazySeq.seq(LazySeq.java:58)
        at clojure.lang.ChunkedCons.chunkedNext(ChunkedCons.java:59)
        at clojure.core$chunk_next.invokeStatic(core.clj:708)
        at clojure.core$reduce1.invokeStatic(core.clj:942)
        at clojure.core$set.invokeStatic(core.clj:4113)
        at clojure.core$set.invoke(core.clj:4105)
        at metabase.driver.sql_jdbc.sync$fast_active_tables.invokeStatic(sync.clj:118)
        at metabase.driver.sql_jdbc.sync$fast_active_tables.doInvoke(sync.clj:108)
        at clojure.lang.RestFn.invoke(RestFn.java:425)
        at metabase.driver.sql_jdbc.sync$fn__72079.invokeStatic(sync.clj:38)
        at metabase.driver.sql_jdbc.sync$fn__72079.invoke(sync.clj:37)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.driver.sql_jdbc.sync$describe_database.invokeStatic(sync.clj:225)
        at metabase.driver.sql_jdbc.sync$describe_database.invoke(sync.clj:221)
        at metabase.driver.sql_jdbc$fn__73838.invokeStatic(sql_jdbc.clj:57)
        at metabase.driver.sql_jdbc$fn__73838.invoke(sql_jdbc.clj:55)
        at clojure.lang.MultiFn.invoke(MultiFn.java:234)
        at metabase.sync.fetch_metadata$fn__52935$db_metadata__52940$fn__52941.invoke(fetch_metadata.clj:13)
        at metabase.sync.fetch_metadata$fn__52935$db_metadata__52940.invoke(fetch_metadata.clj:10)
        at metabase.sync.sync_metadata.tables$fn__54361$db_metadata__54366$fn__54367.invoke(tables.clj:146)
        at metabase.sync.sync_metadata.tables$fn__54361$db_metadata__54366.invoke(tables.clj:143)
        at metabase.sync.sync_metadata.tables$fn__54422$sync_tables_BANG___54427$fn__54428.invoke(tables.clj:163)
        at metabase.sync.sync_metadata.tables$fn__54422$sync_tables_BANG___54427.invoke(tables.clj:158)
        at metabase.sync.util$fn__42218$run_step_with_metadata__42223$fn__42227$fn__42229.invoke(util.clj:345)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:102)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:96)
        at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:119)
        at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:116)
        at metabase.sync.util$fn__42218$run_step_with_metadata__42223$fn__42227.invoke(util.clj:342)
        at metabase.sync.util$fn__42218$run_step_with_metadata__42223.invoke(util.clj:337)
        at metabase.sync.util$fn__42410$run_sync_operation__42415$fn__42416$fn__42417.invoke(util.clj:430)
        at clojure.core$mapv$fn__8445.invoke(core.clj:6912)
        at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
        at clojure.core$reduce.invokeStatic(core.clj:6827)
        at clojure.core$mapv.invokeStatic(core.clj:6903)
        at clojure.core$mapv.invoke(core.clj:6903)
        at metabase.sync.util$fn__42410$run_sync_operation__42415$fn__42416.invoke(util.clj:430)
        at metabase.sync.util$fn__42410$run_sync_operation__42415.invoke(util.clj:424)
        at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480$fn__54481$fn__54482.invoke(sync_metadata.clj:51)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:149)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:145)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
        at clojure.core$partial$fn__5839.invoke(core.clj:2623)
        at metabase.driver$fn__22445.invokeStatic(driver.clj:540)
        at metabase.driver$fn__22445.invoke(driver.clj:540)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at metabase.sync.util$sync_in_context$fn__42130.invoke(util.clj:138)
        at metabase.sync.util$with_db_logging_disabled$fn__42127.invoke(util.clj:129)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:102)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:96)
        at metabase.sync.util$with_start_and_finish_logging$fn__42116.invoke(util.clj:114)
        at metabase.sync.util$with_sync_events$fn__42111.invoke(util.clj:88)
        at metabase.sync.util$with_duplicate_ops_prevented$fn__42102.invoke(util.clj:67)
        at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:164)
        at metabase.sync.util$do_sync_operation.invoke(util.clj:161)
        at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480$fn__54481.invoke(sync_metadata.clj:50)
        at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480.invoke(sync_metadata.clj:47)
        at metabase.sync$fn__54525$sync_database_BANG___54530$fn__54531$fn__54532.invoke(sync.clj:29)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:149)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
        at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:145)
        at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
        at clojure.core$partial$fn__5839.invoke(core.clj:2623)
        at metabase.driver$fn__22445.invokeStatic(driver.clj:540)
        at metabase.driver$fn__22445.invoke(driver.clj:540)
        at clojure.lang.MultiFn.invoke(MultiFn.java:239)
        at metabase.sync.util$sync_in_context$fn__42130.invoke(util.clj:138)
        at metabase.sync.util$with_db_logging_disabled$fn__42127.invoke(util.clj:129)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:102)
        at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:96)
        at metabase.sync.util$with_start_and_finish_logging$fn__42116.invoke(util.clj:114)
        at metabase.sync.util$with_sync_events$fn__42111.invoke(util.clj:88)
        at metabase.sync.util$with_duplicate_ops_prevented$fn__42102.invoke(util.clj:67)
        at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:164)
        at metabase.sync.util$do_sync_operation.invoke(util.clj:161)
        at metabase.sync$fn__54525$sync_database_BANG___54530$fn__54531.invoke(sync.clj:27)
        at metabase.sync$fn__54525$sync_database_BANG___54530.invoke(sync.clj:19)
        at metabase.events.sync_database$process_sync_database_event$fn__74152.invoke(sync_database.clj:38)
        at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: Error : 1424, Position : 195, Sql = SELECT NULL AS table_cat,
       o.owner AS table_schem,
       o.object_name AS table_name,
       o.object_type AS table_type,
       NULL AS remarks
  FROM all_objects o
  WHERE o.owner LIKE :1 ESCAPE '/'
    AND o.object_name LIKE :2 ESCAPE '/'
    AND o.object_type IN ('xxx', 'TABLE', 'VIEW')
  ORDER BY table_type, table_schem, table_name
, OriginalSql = SELECT NULL AS table_cat,
       o.owner AS table_schem,
       o.object_name AS table_name,
       o.object_type AS table_type,
       NULL AS remarks
  FROM all_objects o
  WHERE o.owner LIKE :1 ESCAPE '/'
    AND o.object_name LIKE :2 ESCAPE '/'
    AND o.object_type IN ('xxx', 'TABLE', 'VIEW')
  ORDER BY table_type, table_schem, table_name
, Error Msg = ORA-01424: missing or illegal character following the escape character
```

**To Reproduce**
1 - Using oracle xe docker https://hub.docker.com/r/oracleinanutshell/oracle-xe-11g
2 - CREATE TABLE HR."slash_table/" (
	COLUMN1 VARCHAR2(100)
);
3- Add the database to metabase

**Expected behavior**
There will be no tables on the database and the logs will have the ORA-01424 error

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
Note: It's a blocking bug, because the table that have forward slash are sys tables, not made by us.

**Comment by flamber** [comment]
Possibly related to #12450

**Comment by sbelak** [comment]
This is 2 bugs: the actual error, and the fact that this breaks sync.

**Comment by geraldodev** [comment]
Hi,

Initially I was suspecting the XDB tables because they have forward slashes, but metabase already has this schema flagged to be excluded and it's working.

The offending ones were user names (that represent a schema) on our database.
For the moment I'm stripping them out of the proccess.

```

--- a/src/metabase/driver/sql_jdbc/sync.clj
+++ b/src/metabase/driver/sql_jdbc/sync.clj
@@ -105,6 +105,16 @@
     (mapv #(select-keys % [:table_name :remarks :table_schem])
           (jdbc/result-set-seq rs))))

+(def re-nome-usuario
+  #"i?^(\w+)(\d+)$")
+
+(defn remove-unwanted-schemas
+  [schemas]
+  (->> schemas
+       (remove #(re-find #"/" %) schemas) ;; forward-slash ones
+       (remove #(re-find re-nome-usuario)))) ;; user ones
+
+
 (defn fast-active-tables
   "Default, fast implementation of `active-tables` best suited for DBs with lots of system tables (like Oracle). Fetch
   list of schemas, then for each one not in `excluded-schemas`, fetch its Tables, and combine the results.
@@ -115,7 +125,8 @@
   (with-open [rs (.getSchemas metadata)]
     (let [all-schemas (set (map :table_schem (jdbc/result-set-seq rs)))
           schemas     (set/difference all-schemas (excluded-schemas driver))]
-      (set (for [schema schemas
+      ; (log/info (str "FAST-ACTIVE-TABLES -> schemas: !!" schemas))
+      (set (for [schema (remove-unwanted-schemas schemas)
                  table  (get-tables metadata schema db-name-or-nil)]
              (let [remarks (:remarks table)]
                {:name        (:table_name table)

```

**Comment by sbelak** [comment]
@geraldodev when you say there will be no tables in MB, does this DB have any tables in schemas with names that parse ok? Just trying to understand how severely it breaks sync (when we tried to repo, we only get it to fail for specific tables/schemas).

**Comment by geraldodev** [comment]
The error halted the process of schema identification. Answering the
question, no , there were no table. But it can be because the real data was
on the schemas below.
After changing the code to isolate the ofending schemas , I ended up
deleting the recognized schema, because I've started to see errors of
queries that went wrong when trying to fingerprint the data, similar to a
problem with mysql, on a issue that was opened but was not solved because
of the lack of the information.

Regards,

Geraldo Lopes de Souza

Em qua., 30 de set. de 2020 às 14:33, Simon Belak <notifications@github.com>
escreveu:

**Comment by camsaul** [comment]
This is the same underlying issue as #12450

**Comment by sbelak** [comment]
I'm pretty sure, it's a different issue. The ultimate cause (slashes in name) is the same, but this one explodes in Oracle-land, not with our permissions.

**Comment by flamber** [comment]
Fixed by #13340
